drivers/scheduler/k8s/k8s.go

```diff
@@ -1123,35 +1146,46 @@ func (k *K8s) destroyCoreObject(spec interface{}, opts map[string]bool, app *spe
 
 }
 
-func (k *K8s) createAutopilotObject(apParams *scheduler.AutopilotParameters) (*ap_api.AutopilotRule, error) {
-	obj := &ap_api.AutopilotRule{}
-	apRuleParams := apParams.AutopilotRuleParameters
-	obj.Name = apParams.Name
-	obj.Namespace = apParams.Namespace
-	obj.Labels = map[string]string{"creator": "torpedo"}
-	obj.Spec.PollInterval = apParams.PollInterval
-	obj.Spec.ActionsCoolDownPeriod = apRuleParams.ActionsCoolDownPeriod
-	obj.Spec.Selector.LabelSelector.MatchLabels = apRuleParams.MatchLabels //map[string]string{"name": "pgbench-data"}
-	obj.Spec.NamespaceSelector.LabelSelector.MatchLabels = map[string]string{"creator": "torpedo"}
-	for _, ruleExpression := range apRuleParams.RuleConditionExpressions {
-		exprVolumeUsage := &ap_api.LabelSelectorRequirement{
-			Key:      ruleExpression.Key,
-			Operator: ap_api.LabelSelectorOperator(ruleExpression.Operator),
-			Values:   ruleExpression.Values,
-		}
-		obj.Spec.Conditions.Expressions = append(obj.Spec.Conditions.Expressions, exprVolumeUsage)
+func (k *K8s) addAppSpecEnvironmentVariables(containerEnvVars []v1.EnvVar, appEnvVars []scheduler.EnvVar) []v1.EnvVar {
+
+	var tmpEnvMapVars = make(map[string]string)
+	var envVars []v1.EnvVar
+	for _, cEnvVar := range containerEnvVars {
+		tmpEnvMapVars[cEnvVar.Name] = cEnvVar.Value
 	}
-	for _, ruleAction := range apRuleParams.RuleActions {
-		actions := &ap_api.RuleAction{
-			Name:   ruleAction.Name,
-			Params: ruleAction.Params,
-		}
-		obj.Spec.Actions = append(obj.Spec.Actions, actions)
+	for _, appEnvVar := range appEnvVars {
+		tmpEnvMapVars[appEnvVar.Name] = appEnvVar.Value
+	}
+	for key, value := range tmpEnvMapVars {
+		envVars = append(envVars, v1.EnvVar{
+			Name:  key,
+			Value: value,
+		})
 	}
 
-	logrus.Infof("Using Autopilot Object: %+v\n", obj)
+	return envVars
+}
 
-	return obj, nil
+func (k *K8s) isPvcAnnotationAutopilotEnabled(spec interface{}) bool {
```

Comment:
just called this `isAutopilotEnabledForPVC`. No need to have annotations in the function name. Same for other functions